config/http-headers.js

```diff
@@ -118,4 +118,18 @@ const config = {
 	},
 };
 
+/*
+	The matrix based messenger loads its assets (scripts, styles, fonts, images) from an specified domain.
+	After initialization the chat protocol communicates with its home server.
+ */
+if (Configuration.get('FEATURE_MATRIX_MESSENGER_ENABLED')) {
+	// Messenger assets
+	config.contentSecurityPolicy.corsDefault.defaultSrc += ` ${Configuration.get('MATRIX_MESSENGER_EMBED_URI')}`;
+	config.contentSecurityPolicy.corsDefault.scriptSrc += ` ${Configuration.get('MATRIX_MESSENGER_EMBED_URI')}`;
+	config.contentSecurityPolicy.corsDefault.styleSrc += ` ${Configuration.get('MATRIX_MESSENGER_EMBED_URI')}`;
+	config.contentSecurityPolicy.corsDefault.fontSrc += ` ${Configuration.get('MATRIX_MESSENGER_EMBED_URI')}`;
+	// Chat communication
+	config.contentSecurityPolicy.corsDefault.defaultSrc += ` ${Configuration.get('MATRIX_MESSENGER_HOMESERVER_URI')}`;
+}
+
```

Comment:
Hätte man hier nicht einfach 2 const variablen "homeserver" und "matrixmessenger" benutzen können in dem If-Zweig, diese variablen dann in dem contentSecurityPolicy{..} object an den jeweiligen stellen hinterlegen? Der If-Zweig muss dann natürlich vor dem Object passieren.

Reply:
Natürlich, ich habe es entsprechend geändert.
Ich hatte es nachgelagert implementiert, da es sich durch den if-Zweig um variablen handelt die nicht immer hinzugefügt werden.
Es hat aber auch Vorteil oben gleich alle möglichen Werte in `corsDefault` zu sehen.

Reply:
Ich bin mir nicht sicher ob Inline IF-Zweige (variable ? variable2 : ' ') sinnvoll sind wegen der übersichtlichkeit. Viell. einfach ein IF-Zweig nehmen davor die variablen deklarieren (dann let benutzen anstatt const) und dann in dem IF überprüfen und setzen. Ansonsten sieht es erst einmal gut aus von meiner Seite aus.

Reply:
Ist entsprechend geändert.

Reply:
sehr gut.